karbon/karbon_api.go

```diff
@@ -22,6 +24,69 @@ type Client struct {
 	Meta            MetaService
 }
 
+type ClientService interface {
+	CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error)
+	DeleteK8sRegistration() (*K8sClusterRegistrationDeleteResponse, error)
+	GetK8sRegistration() (*K8sClusterRegistration, error)
+	GetK8sRegistrationList() (*K8sClusterRegistrationList, error)
+}
+
+func (op Client) CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error) {
+	ctx := context.TODO()
+
+	path := "/v1-alpha.1/k8s/cluster-registrations/"
```

Comment:
Should we consider moving the methods for v1alpha1 APIs to a karbon/v1alpha1 directory to be transparent about the fact that these are not v1 APIs?

Reply:
Alternatively, we can consider add `V1Alpha1`, `V1Beta1` to the Karbon client.
```go
karbonClient.V1Aplha1.CreateK8sRegistration(...)
```

Reply:
yes thats the plan as we discussed in the last meeting. lets start doing that with this PR itself

Reply:
this part has been updated.